File: 参考/comp9041-454/9041-labonserver/lab06/all_whales.py
#!/usr/bin/python
from sys import *
from re import *
wi={}
wl={}
while 1:
    try:
        data=raw_input()
        data=data.lower()
        data=sub(' +',' ',data)
        data=sub('^ ','',data)
        data=sub(' $','',data)
        data=sub('s$','',data)
        data=sub(' ','=',data,1)
        wa=data.split('=')[1]
        wn=int(data.split('=')[0])
        if wa not in wi:
            wi[wa]=wn
        else:
            wi[wa]+=wn
        if wa not in wl:
            wl[wa]=1
        else:
            wl[wa]+=1
    except:
        wk=sorted(wi.keys())
        for key in wk:
            print('{} observations: {} pods, {} individuals'.format(key,wl[key],wi[key]))
        exit()

File: 参考/comp9041-454/9041-labonserver/lab06/all_whales.pl
#!/usr/bin/perl -w
while ($data=<STDIN>){
   $data=~ tr/A-Z/a-z/;
   $data=~ s/ +/ /g;
   $data=~ s/^ //;
   $data=~ s/ $//;
   $data=~ s/s$//;
   $data=~ s/ /=/;
   $wa=(split /=/,$data)[1];
   chomp $wa;
   $wn=(split /=/,$data)[0];
   $wi{$wa}+=$wn;
   $wl{$wa}++;
}
foreach $wa(sort keys %wi){
   printf "$wa observations: $wl{$wa} pods, $wi{$wa} individuals\n";
}
